src/gimelstudio/corenodes/filter/opacity_node/opacity_node.py

```diff
@@ -0,0 +1,85 @@
+# ----------------------------------------------------------------------------
+# Gimel Studio Copyright 2019-2021 by Noah Rahm and contributors
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+# http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+# ----------------------------------------------------------------------------
+
+
+from gimelstudio import api
+
+
+class OpacityNode(api.Node):
+    def __init__(self, nodegraph, _id):
+        api.Node.__init__(self, nodegraph, _id)
+
+    @property
+    def NodeMeta(self):
+        meta_info = {
+            "label": "Opacity",
+            "author": "Gimel Studio",
+            "version": (0, 0, 1),
+            "category": "FILTER",
+            "description": "",
+        }
+        return meta_info
+
+    def NodeInitProps(self):
+        self.value = api.PositiveIntegerProp(
+            idname="Opacity Value",
+            default=25,
+            min_val=0,
+            max_val=100,
+            widget=api.SLIDER_WIDGET,
+            label="Opacity:",
+        )
+        self.NodeAddProp(self.value)
+
+    def NodeInitParams(self):
+        image = api.RenderImageParam('Image')
+        self.NodeAddParam(image)
+
+    def NodeEvaluation(self, eval_info):
+        image1 = self.EvalParameter(eval_info, 'Image')
+        opacity_value = self.EvalProperty(eval_info, 'Opacity Value')
+
+        render_image = api.RenderImage()
+
+        # Make correction for slider range of 1-100
+        opacity_value = (opacity_value * 0.01)
+
+        shader = """
+        #version 330
+
+        uniform sampler2D image;
+        out vec4 out_color;
+        uniform float opacityValue;
+
+        void main() {
+            vec4 color = texelFetch(image, ivec2(gl_FragCoord.xy), 0);
+            out_color = vec4(color.r, color.g, color.b, opacityValue);
+        }
+        """
```

Comment:
> Figure out loading GLSL from a file rather than hardcoding it in the python file

I don't think it should be any harder than something like the following.

```python
with open("./gimelstudio/corenodes/filter/opacity_node/opacity.glsl", "r") as f:
    shader = f.read()
```

Reply:
That was my original solution (see the method in the ``GLSLRenderer`` class) however the path seems to get messed up depending on how you run the script. Maybe since it is relative path based?

Reply:
> the path seems to get messed up depending on how you run the script

That is the reason. With Python, relative paths are added onto the absolute path of the current working directory (unless you change it, this will be the directory containing the Python file you ran). For example, if you start a python program with

```bash
cd ~/GimelStudio/src
python3 main.py
```

then the absolute path of the relative path ``./gimelstudio/corenodes/filter/opacity_node/opacity.glsl`` would be ``~/GimelStudio/src/gimelstudio/corenodes/filter/opacity_node/opacity.glsl``

Following on from that example, if you were to run the python program with

```bash
cd ~/GimelStudio
python3 src/main.py
```
then the absolute path of the relative path ``./gimelstudio/corenodes/filter/opacity_node/opacity.glsl`` would be ``~/GimelStudio/gimelstudio/corenodes/filter/opacity_node/opacity.glsl``. As the file does not exist, this raises a ``FileNotFoundError``.

*NOTE: I've used ``~`` to shorten ``/home/$USERNAME$/``*

Reply:
Ah, yes. Well, I wonder how that can be solved....Maybe use the ``APP_DIR`` variable from ``constants.py``?

Reply:
Where/how do you want to run ``main.py``? If you really want to follow something similar to the second example then you should be able to use ``os.chdir()`` to solve this problem